.ci/testpub.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+pub_to_test=$1
+
+TMP=$(mktemp --suffix=testpub -d)
+cleanup_tmp () {
+  rm -Rf $TMP
+}
+trap cleanup_tmp EXIT
```

Comment:
Did not know about trap statements, this is a really excellent practice to make sure you cover your tracks.

Reply:
I'm fairly certain every programming language has some sort signal handler, for example 
[node](https://nodejs.org/api/process.html).  Mitchell Hashimoto has said this one of the things he had screwed up initially with vagrant, by adding signal handling it became a much more robust application.